clean-modular-code/activity-3/clean-code-activity-3.md

```diff
@@ -117,113 +142,71 @@ def load_clean_json(file_path, columns_to_keep):
 ```
 
 +++ {"editable": true, "slideshow": {"slide_type": ""}}
 
-## Activity 1: Part 1
+## Activity 3, part 2: Add checks to the `format_date` function
 
-The code below creates a `pd.DataFrame` with the first 15 publications in the JOSS sample data.json file. This is the first of 3 files that you need to process in your workflow.
+The code below creates a `pd.DataFrame` with the first 15 publications in the JOSS sample `data.json` file. This is the first of 3 files you must process in your workflow.
 
-Your first task is to process and reformed the published_date column in the data to make it a `pandas.datetime` object.H aving a date in a useful format will allow you to do further analysis later such as counting publications by month and year! 
+Your first task is to process and format the `published_date `column in the data to make it a `pandas.datetime` object. Having a date in a `datetime` format will allow you to do time-related analysis on your data, such as counting publications by month and year! The expected CrossRef published date should is below:
 
-The function that you should focus on here is:
+```json
+"published": {
+      "date-parts": [
+        [
+          2022,
+          11,
+          27
+        ]
+      ]
+    }
+```
+
+However, the date is not always formatted as expected in the above sample data.
+
+For this activity, focus on adding checks to the `format_date` function. **IMPORTANT:** Use the sample data provided below for your troubleshooting exercise. This will allow you to focus on fixing only one function rather than trying to troubleshoot the entire workflow!  
 
 
+:::{admonition} Activity 2: Part 2
+:class: warning 
+
 In small groups, do the following:
 
 1. Evaluate the `published_date` field in the data created below and answer the question: 
 
-* Do you see any unusually formatted values that may be responsible for making your code above fail?
-
-:::{tip}
-Using `DataFrame.head(7)` and `DataFrame.tail(number_of_rows_here)` will be helpful as you troubleshoot the data. 
-:::
+* Do you see any unusually-formatted values that may be responsible for making your code above fail?
 
 2. Once you have a list of issues you observe in the data, address them by modifying the `format_date` function below.
+:::
 
 +++ {"editable": true, "slideshow": {"slide_type": ""}, "tags": ["hide-output", "hide-cell"]}
 
-In this activity, you will be working with Pandas `DataFrames`. You may find the `.apply` function to be particularly useful. 
+###  Format dates with `pandas.datetime`
 
-### How to apply functions to DataFrame values: `pandas.apply()` 
-
-The `pandas.apply()` method allows you to apply any function to rows or columns in a `pandas.DataFrame`. For example, you can use it to perform operations on specific column or row values. When you use `.apply()`, you can specify whether you want to apply the function across columns `(axis=0)` (the default) or across rows `(axis=1)`. 
+What Does `02d` Mean?
```

Comment:
The ordering here is throwing me off somewhat. I haven't yet seen `02d` so I don't have the question: what does it mean? It might be good so start this section with the f-string example, isolated.

Also, it might be good to link out to the format ["mini-language"](https://docs.python.org/3/library/string.html#formatspec). It is very powerful, too much to be covered here, but could be an extra learning resource.

Reply:
Added a suggestion for the "mini-language"